modules/social_features/social_tagging/social_tagging.module

```diff
@@ -582,9 +583,15 @@ function social_tagging_form_views_exposed_form_alter(&$form, FormStateInterface
  *   A node.
  *
  * @return mixed|string
- *   A machine name so it can be used pogrammatically.
+ *   A machine name so it can be used programmatically.
+ *
+ * @deprecated in social:11.3.10 and is removed from social:12.0.0. Use
+ *   \Drupal::service('social_core.machine_name')->transform() instead.
+ *
+ * @see https://www.drupal.org/node/3306214
  */
 function social_tagging_to_machine_name($text) {
+  @trigger_error('social_tagging_to_machine_name() is deprecated in social:11.3.10 and is removed from social:12.0.0. Use "\Drupal::service(\'social_core.machine_name\')->transform()" instead. See https://www.drupal.org/node/3306214', E_USER_DEPRECATED);
   return preg_replace('@[^a-z0-9-]+@', '_', strtolower($text));
```

Comment:
Shouldn't this be replaced with the new service? Because we want people to use the new fix, but still give them the option to use this function name.

When checking documentation here
https://www.drupal.org/about/core/policies/core-change-policies/drupal-deprecation-policy#how-function

where they do something similar.

Reply:
This is exactly what I did in the first PR https://github.com/goalgorilla/open_social/pull/3082 but then it means it landed in 11.5.0, so we reverted and chose to not alter the function body and replace all the calls to make it land in a patch release version.

Reply:
Sorry I am incorrect. As the function changes the outcome it would be indeed backwards incompatible by doing what I said.